README.tfdoc.hcl

```
@@ -0,0 +1,826 @@
+      - `aws_ecs_task_definition`
+    END
+  }
+
+  section {
+    title   = "Getting Started"
+    content = <<-END
+      Most common usage of the module:
+
+      ```hcl
+      module "terraform-aws-ecs-service" {
+        source = "git@github.com:mineiros-io/terraform-aws-ecs-service.git?ref=v0.0.1"
+
+        name   = "name-of-service"
+        cluser = "name-of-cluster"
+
+        task_definition = {
+          network_mode = "awsvpc"
+          cpu          = 256
+          memory       = 512
+
+          container_definitions = jsonencode([{
+            name      = "nginx"
+            image     = "nginx:latest"
+            essential = true
+          }])
+        }
+      }
+      ```
+    END
+  }
+
+  section {
+    title   = "Module Argument Reference"
+    content = <<-END
+      See [variables.tf] and [examples/] for details and use-cases.
+    END
+
+    section {
+      title = "Main Resource Configuration"
+
+      variable "name" {
+        required    = true
+        type        = string
+        description = <<-END
+          Name of the service (up to 255 letters, numbers, hyphens, and
+          underscores).
+        END
+      }
+
+      variable "cluster" {
+        required    = true
+        type        = string
+        description = <<-END
+          The Amazon Resource Name (ARN) of the ECS Cluster where this service
+          should run.
+        END
+      }
+
+      variable "launch_type" {
+        type        = string
+        default     = "EC2"
+        description = <<-END
+          Launch type that will be used to run your service. The valid values
+          are `EC2`, `FARGATE`, and `EXTERNAL`.
+        END
+      }
+
+      variable "network_configuration" {
+        type        = object(network_configuration)
+        description = <<-END
+          Network configuration for the service. This parameter is required for
+          task definitions that use the awsvpc network mode to receive their
+          own Elastic Network Interface, and it is not supported for other
+          network modes.
+        END
+
+        attribute "subnets" {
+          required    = true
+          type        = set(string)
+          description = <<-END
+            Subnets associated with the task or service.
+          END
+        }
+
+        attribute "security_groups" {
+          type        = set(string)
+          description = <<-END
+            Security groups associated with the task or service. If you do not
+            specify a security group, the default security group for the VPC is
+            used.
+          END
+        }
+
+        attribute "assign_public_ip" {
+          type        = bool
+          description = <<-END
+            Assign a public IP address to the ENI (Fargate launch type only).
+            Valid values are `true` or `false`.
+          END
+        }
+      }
+
+      variable "capacity_provider_strategy" {
+        type        = set(capacity_provider_strategy)
```

```suggestion
        type        = list(capacity_provider_strategy)
```

This is a list afaik

---

@soerenmartius The `schema.json` has nesting mode as `set`:
```
"block_types": {
                            "capacity_provider_strategy": {
                                "nesting_mode": "set",
                                "block": {
                                    "attributes": {
                                        "base": {
                                            "type": "number",
                                            "description_kind": "plain",
                                            "optional": true
                                        },
                                        "capacity_provider": {
                                            "type": "string",
                                            "description_kind": "plain",
                                            "required": true
                                        },
                                        "weight": {
                                            "type": "number",
                                            "description_kind": "plain",
                                            "optional": true
                                        }
                                    },
                                    "description_kind": "plain"
                                }
                            },
```

---

gotcha thanks!